scipy/special/_generate_pyx.py

```diff
@@ -1476,10 +1473,25 @@ def main():
     for f, sig in functions.items():
         ufuncs.append(Ufunc(f, sig))
         fused_funcs.append(FusedFunc(f, sig))
-    generate_ufuncs("_ufuncs", "_ufuncs_cxx", ufuncs)
-    generate_ufuncs_type_stubs("_ufuncs", ufuncs)
-    generate_fused_funcs("cython_special", "_ufuncs", fused_funcs)
+    generate_ufuncs(os.path.join(outdir, "_ufuncs"),
+                    os.path.join(outdir, "_ufuncs_cxx"),
+                    ufuncs)
+    generate_ufuncs_type_stubs(os.path.join(outdir, "_ufuncs"),
+                               ufuncs)
+    generate_fused_funcs(os.path.join(outdir, "cython_special"),
+                         os.path.join(outdir, "_ufuncs"),
+                         fused_funcs)
 
 
 if __name__ == "__main__":
-    main()
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-o", "--outdir", type=str,
+                        help="Path to the output directory")
+    args = parser.parse_args()
+
+    if not args.outdir:
+        raise ValueError(f"Missing `--outdir` argument to _generate_pyx.py")
+
+    outdir_abs = os.path.join(os.getcwd(), args.outdir)
+
+    main(outdir_abs)
```

Comment:
I am thinking that why we need changes in this file. Using `setuptools` the original file works, so probably this should have worked as it is in `meson.build`. Simply put, is it compulsory to do changes here?

Reply:
It is, because what this script was doing was write new files in-tree, and this is something Meson forbids - all generated files must go into the build directory. The reason for that is that you can then have multiple build directories, which is unreliable if you allow writing in-tree.

The only way around this restriction is hacking it like you did before with `run_command`; the only reason that happens to work is that then part of the build is already run at configure time, so the files exist already when Meson first checks for them. But this messes with caching and is unsupported so could break in the future.

Reply:
> all generated files must go into the build directory

Ideally, this should be the case i.e., due to build source tree shouldn't be affected. I think it is great that these changes are made.

Reply:
> P.S. In fact, I am thinking, this can be done with `_generate_pyx.py` files in other modules too.

Yes, that's already the case for some - I copied the approach from `linalg/_generate_pyx.py`.